spec/models/event_schedule_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe EventSchedule, type: :model do
+  describe '.populate_upcoming_show_and_tell!' do
+    subject { described_class.populate_upcoming_show_and_tell! }
+    before { travel_to Date.new(2017, 1, 1) }
+    after { travel_back }
+
+    let(:fridays) do
+      counter = 0
+      dates = []
+      4.times do
+        dates << Date.new(2017, 1, 6 + counter)
+        counter += 7
+      end
+      dates
+    end
+
+    it 'creates show and tell events' do
+      expect { subject }
+        .to change { Event.where(name: 'Show & Tell').length }
+    end
+
+    it 'creates show and tell events that take place on Friday' do
+      subject
+      Event.where(name: 'Show & Tell').each do |event|
```

Comment:
Dang man this recommendations are on point. Learning a lot about methods, ways of doing things that I didn't already know.